classes/PrestaShopAutoload.php

```diff
@@ -286,9 +286,10 @@ protected function getClassesFromDir($path, $hostMode = false)
                 } elseif (substr($file, -4) == '.php') {
                     $content = file_get_contents($rootDir . $path . $file);
 
-                    $namespacePattern = '[\\a-z0-9_]*[\\]';
-                    $pattern = '#\W((abstract\s+)?class|interface)\s+(?P<classname>' . basename($file, '.php') . '(?:Core)?)'
-                                . '(?:\s+extends\s+' . $namespacePattern . '[a-z][a-z0-9_]*)?(?:\s+implements\s+' . $namespacePattern . '[a-z][\\a-z0-9_]*(?:\s*,\s*' . $namespacePattern . '[a-z][\\a-z0-9_]*)*)?\s*\{#i';
+                    $namePattern = '[a-z_\x7f-\xff][a-z0-9_\x7f-\xff]*';
```

Comment:
Why adding \x7f and \xff, I don't find any reason to add it?

Reply:
https://stackoverflow.com/questions/6362241/php-variable-function-class-names-using-special-characters#6362268 To match all valid names no matter if Unicode names are currently used or not.